File: sqlfs#/SqlFsTransaction.cs
namespace com.sss.sqlfs
{

	using SQLiteDatabase = android.database.sqlite.SQLiteDatabase;
	using com.sss.sqlfs.helper;

	internal class SqlFsTransaction : IDisposable
	{
	   private SQLiteDatabase db;

	   internal SqlFsTransaction(SQLiteDatabase db)
	   {
		   this.db = db;
		   this.db.beginTransaction();
	   }

	   internal virtual void fsOpSuccess()
	   {
		   db.setTransactionSuccessful();
	   }

	   public virtual void dispose()
	   {
		   db.endTransaction();
	   }
	}

}

File: SqlFs/src/com/sss/sqlfs/SqlFsTransaction.java
package com.sss.sqlfs;

import android.database.sqlite.SQLiteDatabase;
import com.sss.sqlfs.helper.*;

class SqlFsTransaction implements IDisposable
{
   private SQLiteDatabase db;
   
   SqlFsTransaction(SQLiteDatabase db)
   {
	   this.db = db;
	   this.db.beginTransaction();
   }
   
   void fsOpSuccess()
   {
	   db.setTransactionSuccessful();
   }
	
   public void dispose()
   {
	   db.endTransaction();
   }
}
